platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/results/TestListenerInternal.java

```diff
@@ -24,8 +24,10 @@
 import org.gradle.internal.scan.UsedByScanPlugin;
 import org.gradle.internal.service.scopes.EventScope;
 import org.gradle.internal.service.scopes.Scope;
+import org.gradle.internal.service.scopes.StatefulListener;
 
 @UsedByScanPlugin
+@StatefulListener
```

Comment:
This annotation allows the listener implementations to be automatically wired. Services that implement this interface will be auto-registered to receive events from the listener manager without explicit wiring.

For events to be received in parallel, the actual listener _implementation_ would need to be annotated with `@ParallelListener`. Since it is not, we still perform locking in the listener manager when calling into the listener implementation.